src/Serilog/Core/Pipeline/MessageTemplateCache.cs

```diff
@@ -36,9 +36,10 @@ public MessageTemplate Parse(string messageTemplate)
 
         // ReSharper disable once InconsistentlySynchronizedField
         // ignored warning because this is by design
-        var result = (MessageTemplate?)_templates[messageTemplate];
-        if (result != null)
+        if (_templates.TryGetValue(messageTemplate, out var result))
+        {
```

Comment:
Thread safety guaranties for two data structures are different:
[Dictionary<,>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-7.0#thread-safety):

> A [Dictionary<TKey,TValue>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-7.0) can support multiple readers concurrently, as long as the collection is not modified.

[Hashtable](https://learn.microsoft.com/en-us/dotnet/api/system.collections.hashtable?view=net-7.0#thread-safety):
> It is thread safe for multi-thread use when only **one of the threads perform write** (update)

That was the main motivation to use `Hashtable`, otherwise read operation should have been placed into the `lock` section.

Reply:
@skomis-mm i did not know that. thanks so much for the info

Reply:
@nblumhardt do u want a PR with a comment containing the above info?

Reply:
Awesome idea, thanks :-)

Reply:
@nblumhardt  https://github.com/serilog/serilog/pull/1828